### ISSUE TITLE
Parameter `--require` cannot load an ES module.

### ISSUE THREAD

**Issue by regseb** [mochajs/mocha#4281]
### Prerequisites

- [x] Checked that your issue hasn't already been filed by cross-referencing [issues with the `faq` label](https://github.com/mochajs/mocha/issues?utf8=%E2%9C%93&q=is%3Aissue%20label%3Afaq%20)
- [ ] Checked next-gen ES issues and syntax problems by using the same environment and/or transpiler configuration without Mocha to ensure it isn't just a feature that actually isn't supported in the environment in question or a bug in your code.
- [x] 'Smoke tested' the code to be tested by running it outside the real test suite to get a better sense of whether the problem is in the code under test, your usage of Mocha, or Mocha itself
- [x] Ensured that there is no discrepancy between the locally and globally installed versions of Mocha. You can find them with: `node node_modules/.bin/mocha --version`(Local) and `mocha --version`(Global). We recommend that you _not_ install Mocha globally.

### Description

Parameter `--require` cannot load an ES module.

### Steps to Reproduce

- download [_testcase.zip_](https://github.com/mochajs/mocha/files/4612189/testcase.zip) which contains 3 files:

> _package.json_:

```json
{
    "name": "testcase",
    "version": "1.0.0",
    "type": "module",
    "scripts": {
        "test": "mocha test.js"
    },
    "dependencies": {
        "mocha": "^7.1.2"
    },
    "mocha": {
        "require": "require.js"
    }
}
```

> _require.js_:

```javascript
// An ES module because { "type": "module" } in package.json.
```

> _test.js_:

```javascript
import assert from "assert";

describe("foo", function () {
    it("bar", function () {
        assert.ok("baz");
    });
});
```

- unzip _testcase.zip_;
- `npm install`
- `npm test`

**Expected behavior:** [What you expect to happen]

```
> testcase@1.0.0 test /home/regseb/testcase
> mocha test.js



  foo
    ✓ bar


  1 passing (6ms)
```
**Actual behavior:** [What actually happens]

```
> testcase@1.0.0 test /home/regseb/testcase
> mocha test.js

mocha inspect [spec..]

Run tests with Mocha

Rules & Behavior
  --allow-uncaught           Allow uncaught errors to propagate        [boolean]
[...]
  --list-reporters   List built-in reporters & exit                    [boolean]

✖ ERROR: Must use import to load ES Module: /home/regseb/testcase/require.js
require() of ES modules is not supported.
require() of /home/regseb/testcase/require.js from /home/regseb/testcase/node_modules/mocha/lib/cli/run-helpers.js is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules.
Instead rename require.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from /home/regseb/testcase/package.json.

npm ERR! Test failed.  See above for more details.
```

**Reproduces how often:** [What percentage of the time does it reproduce?] 100%

### Versions

- The output of `mocha --version`: _zsh: command not found: mocha_
- The output of `node node_modules/.bin/mocha --version`: _7.1.2_
- The output of `node --version`: _v13.14.0_
- Your operating system
  - name and version: _Ubuntu 19.10_
  - architecture (32 or 64-bit): _64-bit_
- Your shell (e.g., bash, zsh, PowerShell, cmd): _zsh 5.7.1 (x86_64-ubuntu-linux-gnu)_
- Your browser and version (if running browser tests): N.A.
- Any third-party Mocha-related modules (and their versions): no
- Any code transpiler (e.g., TypeScript, CoffeeScript, Babel) being used (and its version): no

### Additional Information

Fix proposal: replace [`require`](https://github.com/mochajs/mocha/blob/master/lib/cli/run-helpers.js#L89) by [`import`](https://nodejs.org/api/esm.html#esm_import_expressions). But _dynamic import_ has been added to [Node _13.2.0_](https://wiki.developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/import#Browser_compatibility).

**Comment by boneskull** [comment]
This is known, and is a "missing feature"

**Comment by JacobLey** [comment]
Ran into this issue as well.

In a quick dive through the code, this seems like a feasible and quick tweak to support. I am happy to take a swing at a PR, but I wanted to discuss possible reasons not to support this, or alternative solutions.

The error spawns here: https://github.com/mochajs/mocha/blob/master/lib/cli/run-helpers.js#L92 See how the require method is being used directly.

If we swapped that require with the existing requireOrImport it should auto-magically support ESM.

Of course that makes the "require" async, but considering the handleRequires already has an async signature, there would not need to be any change to that methods usage. Would just need to convert that .reduce() into something async friendly (for ... of?).
